### ISSUE TITLE
import fails with 'no default export'

### ISSUE THREAD

**Issue by StokeMasterJack** [Microsoft/TypeScript-React-Starter#8]
This import works fine under babel:

`import React from "react";`

But under TypeScript it complains of:
(1,8): error TS1192: Module '"/Volumes/repos/tc-web-ts/node_modules/@types/react/index"' has no default export.

**Comment by snerks** [comment]
Have you tried:

`import * as React from 'react';`

**Comment by StokeMasterJack** [comment]
Yes. But I have hundreds of files like this. I thought TypeScript was a superset of JavaScript?

**Comment by bluetech** [comment]
Add `"allowSyntheticDefaultImports": true` to your `tsconfig.json` if you want it to work like this.

**Comment by nitishdayal** [comment]
^ + IME change tsconfig.json > compilerOptions.module to 'es2015'. That's the configuration I've been using on my React/TS projects, allows for the usual default import syntax that most have become accustomed to.

**Comment by StokeMasterJack** [comment]
> Add "allowSyntheticDefaultImports": true to your tsconfig.json if you want it to work like this.

This did not work. Now I am getting this error:

`Uncaught TypeError: Cannot read property 'render' of undefined`

Here is the entire app:

```
import React from "react";
import ReactDOM from "react-dom";

function render() {
  ReactDOM.render(
    <h1>Hello</h1>,
    document.getElementById("root")
  );
}
render();
```

**Comment by DanielRosenwasser** [comment]
Since Webpack will do the "correct" import for you, feel free to use `import React from "react"` and use

```json
"target": "es5",
"module": "es2015",
"moduleResolution": "node"
```

**Comment by DanielRosenwasser** [comment]
This has to do with the CommonJS/ES module interop, which is not specified in the ECMAScript standard. In the near future, I believe that Babel and TypeScript will be more aligned on this.

**Comment by StokeMasterJack** [comment]
The funny thing is that these are all runtime errors.

It's a bit ironic that after converting my code base to TypeScript, I'm now plagued with _runtime_ errors.

**Comment by Aqours** [comment]
Same isuue.
> @bluetech `"allowSyntheticDefaultImports": true` to your `tsconfig.json` if you want it to work like this.
This is temporary solution.
I think it's a bug when coding with JS and TS.

**Comment by HyunSeob** [comment]
Is there a plan for supporting this issue?

**Comment by wtilton** [comment]
I just had the same error, this was the first hit on a google search. Was strange to me as I was importing above just fine and then saw the difference:

import User from "user";

had the error. This:

import {User} from "user";

did not. Might help someone else.

**Comment by wk-j** [comment]
Try this `"esModuleInterop": true`

**Comment by ZhouYK** [comment]
@wk-j it works.thanks

**Comment by Mnuzz** [comment]
@wk-j Your solution breaks my import of moment.js.

**Comment by wk-j** [comment]
@Mnuzz You can import moment like this

```
import moment from "moment"
```

**Comment by Mnuzz** [comment]
Apparently not. I'll set up a sample project and show you what happens.

**Comment by jynweythek223** [comment]
Neither "allowSyntheticDefaultImports" nor "esModuleInterop" works in my case. Tried fresh CRA with react-scripts-ts and setting up from scratch with Typescript 2.7.2.

**Comment by felixrabe** [comment]
@Mnuzz did you get around to put together the sample project?

**Comment by gogoyqj** [comment]
need both
```
    "allowSyntheticDefaultImports": true,
    "esModuleInterop": true
````

**Comment by haseose** [comment]
@gogowings   Thanks it's work.

**Comment by Jun711** [comment]
@bluetech @gogoyqj 
set 
"allowSyntheticDefaultImports": true
within compilerOptions?

**Comment by hannah-hpcnt** [comment]
@Jun711 If your module is `esnext`, try opting in `"moduleResolution": "node"` in your tsconfig. ([Reference](https://github.com/Microsoft/TypeScript/issues/16820))

**Comment by hamza00111** [comment]
Try this :
import {React} from "react";

**Comment by MartinYounghoonKim** [comment]
@gogoyqj  Thanks. I solved this problem.

I hope my config helps other.
`
  "compilerOptions": {
    "sourceMap": true,
    "module": "esnext",
    "moduleResolution": "node",
    "target": "es5",
    "jsx": "react",
    "allowSyntheticDefaultImports": true,
    "esModuleInterop": true,
    "typeRoots": [
      "./node_modules/@types/"
    ],
`

**Comment by gpresland** [comment]
Following the README, you are supposed to make the following changed in `tslint.json`:

```diff
{
-  "extends": ["tslint:recommended", "tslint-react", "tslint-config-prettier"],
+  "extends": [],
+  "defaultSeverity": "warning",
   "linterOptions": {
     "exclude": [
       "config/**/*.js",
       "node_modules/**/*.ts"
     ]
   }
 }
```
from https://github.com/Microsoft/TypeScript-React-Starter#overriding-defaults

You can also simply add:
```diff
+ "defaultSeverity": "warning",
```
And the issue should go simply be a warning in the console.

**Comment by EvgenyW3** [comment]
Combination of `    
"allowSyntheticDefaultImports": true,
 "esModuleInterop": true
` and in addition `"noImplicitAny": false` did the trick for me, but I don't really like this solution
